server/resp/src/main/java/org/infinispan/server/resp/commands/list/RPUSHX.java

```diff
@@ -0,0 +1,71 @@
+package org.infinispan.server.resp.commands.list;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.infinispan.commons.CacheException;
+import org.infinispan.multimap.api.embedded.MultimapCache;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.util.concurrent.CompletionStages;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @link https://redis.io/commands/rpushx/
+ * Inserts specified values at the tail of the list stored at key, only if key already exists and holds a list.
+ * In contrary to {@link RPUSH} no operation will be performed when key does not yet exist.
+ * Integer reply: the length of the list after the push operation.
+ * @since 15.0
+ */
+public class RPUSHX extends RespCommand implements Resp3Command {
+   public RPUSHX() {
+      super(-3, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+                                                      ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+
+      if (arguments.size() < 2) {
+         // ERROR
+         RespErrorUtil.wrongArgumentNumber(this, handler.allocatorToUse());
+         return handler.myStage();
+      }
+
+      MultimapCache<byte[], byte[]> multimapCache = handler.getMultimap();
+      byte[] key = arguments.get(0);
+
+      CompletableFuture<Boolean> containsKey =
+            multimapCache.containsKey(key);
+      return CompletionStages.handleAndCompose(containsKey, (exists, t1) -> {
+         if (exists) {
+            CompletableFuture<Void> asyncPut = multimapCache.put(key, arguments.get(1));
+            for (int i = 2; i < arguments.size(); i++) {
+               final int nextValue = i;
+               asyncPut = asyncPut.thenCompose(v -> multimapCache.put(key, arguments.get(nextValue)));
+            }
+
+            return CompletionStages.handleAndCompose(asyncPut, (r, t2) -> {
+               if (t2 instanceof CacheException) {
+                  // TODO: test the cache exception message type ?
```

Comment:
Yeah, once we start dealing with more structures, I think we can move this error handling into something centralized.